### ISSUE TITLE
Svelte 5 upgrade (NPM only, no migration)

### ISSUE THREAD

**Issue by rmunn** [sillsdev/languageforge-lexbox#1174]
**Describe the feature**
We want to try migrating to Svelte 5, but the following projects don't support Svelte 5 yet:

- <del>svelte-intl-precompile ([open issue](https://www.github.com/cibernox/svelte-intl-precompile/issues/67) for Svelte 5 support, no PR yet)</del> Supports Svelte 5 just fine since stores are not going away in Svelte 5; the peer dependency error from PNPM is safe to ignore.

That's it! The other libraries we use have released a Svelte 5 version (in some cases requiring a major version bump), and svelte-intl-precompile safely supports Svelte 5 despite the peer dependency warning. So we've decided to try upgrading to use the Svelte 5 NPM packages (including upgrading svelte-check, vite-plugin-svelte, and so on). Then we'll see if anything breaks, and we'll get some practice upgrading components to use runes one at a time before we run `pnpx svelte-migrate svelte-5`.

**Comment by hahn-kev** [comment]
Does svelte-intl-precompile not work in Svelte 5? Sure it uses stores, but that should work fine.

**Comment by rmunn** [comment]
Yes, it should work, but you'll get a warning from pnpm because it depends on precompile-intl-runtime, which in turn declares peer dependencies on `"svelte": "^3.37.0 || ^4.0.0"`. precompile-intl-runtime doesn't yet have an issue open for adding Svelte 5 to its peer dependencies list. So until we've tested it, I plan to leave svelte-intl-precompile on the list — for now, at least. (If it does work, we should file an issue with precompile-intl-runtime to let the developer know that it works with Svelte 5 on at least our codebase).

**Comment by rmunn** [comment]
We decided to upgrade to the Svelte 5 NPM package and check if svelte-intl-precompile really is a blocker, and it turns out it's not. So this issue isn't quite named correctly anymore.
